### ISSUE TITLE
Super, Alt, Ctrl are captured on Linux Server

### ISSUE THREAD

**Issue by ghost** [input-leap/input-leap#936]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/936
> 
> Issue created on: 2020-11-02 by @siavashs
> Issue last updated on: 2021-01-11

**Describe the bug**
I'm using Arch Linux as server and macOS as client, mouse and keyboard control work normally except Super, Alt, Ctrl keys which must trigger Cmd, Alt, Ctrl on macOS.

**To Reproduce**

1. Install Barrier on Server(Linux) and client(macOS)
2. Configure the server & client and start them
3. Move the mouse pointer to client screen(macOS) press Super+Tab buttons
4. The application switch happens on GNOME Shell instead of macOS desktop

**Expected behavior**
Super, Alt, Ctrl must be captured and sent to the client

**Desktop:**
- OS: Linux(server), macOS(client)
- Barrier version 2.3.3

---

> Commented on: 2020-11-10 by @claudioap

Is your server using wayland? Wayland is unsupported and there's a bounty in #109 

---

> Commented on: 2020-11-10 by @siavashs

Correct, I assumed Wayland is supported since it's the default in many distros now.

---

> Commented on: 2021-01-11 by @vii

is the bounty for a Wayland client? The issue here is passing through these special keys from the server.

Possible approaches that I can see
- integrate into the Wayland stack (mutter/gnome-shell) and cooperate with them to capture these events
- get into lower level Linux input layer evemu-record /dev/input/event4

This is separate from implementing the client on Wayland

**Comment by metal3d** [comment]
Fedora 36 on both server and client, same problem.

**Comment by metal3d** [comment]
Correction, only Alt+Tab and Super seems to fail.

**Comment by shymega** [comment]
Noted, thanks for update.

**Comment by ryanwohara** [comment]
Ubuntu 22.04, same `alt+tab` and `super` issues

**Comment by shymega** [comment]
> Ubuntu 22.04, same `alt+tab` and `super` issues

We already know of that issue. Please avoid, in future, from duplicating comments on trackers. To tidy up this issue, I will hide your comment as duplicate. If there's no updates on an issue, it likely hasn't been fixed.

**Comment by shymega** [comment]
To address the downvote - I don't intend to trivialize your input, but at the same time, we need to avoid having dupe comments on a thread. I suppose I acted a bit hastily, as to be fair, the previous comment was about Fedora.

Something I do need to verify though - @metal3d / @ryanwohara - are you running Input Leap on Wayland or X11? We only support X11 officially. If you are running on Wayland, please switch to X11, then test, and report back. Thank you!

**Comment by ryanwohara** [comment]
I'm running X11 on Ubuntu 22.04.

I just got this working after downgrading my kernel to `5.15.0-25-generic` from the previous `5.19.0-46-generic`. Not sure what the takeaway from that is, or if the actual solution was something unrelated that I missed.

**Comment by shymega** [comment]
Interesting. So, the issue is related to X11. Wayland remains to be seen.

I can't see why downgrading the kernel would fix the issue - how bizarre. When you do your next update, can you let us know if it works again? Are you building from Git?

**Comment by ryanwohara** [comment]
Yes, I am building from git. I am on `6446288aa619133f287cf3166bc924a99d073ea2`.

**Comment by shymega** [comment]
OK. So, I wouldn't exactly recommend building from Git, because we still have a few bugs present. However, v3.0.0 is still pending to be released, so I understand.

Did you enable `libei` with the CMake flag?

**Comment by ryanwohara** [comment]
I ran `clean_build.sh` with no flags.

I should also note I installed Nvidia drivers today.

**Comment by shymega** [comment]
OK, so we can rule out this being Wayland contamination.

What are your specs? Do you have an iGPU? I can't see this being related to NVIDIA, I'm thinking it could be evdev.

**Comment by shymega** [comment]
I see, OK. So I don't think it's `evdev`, as that's only when IL is compiled with `libei` support.

I want to put this issue on hold for a bit, at least, until @metal3d reports back. I would like to make sure it's definitely X11.

**Comment by metal3d** [comment]
Oh sorry, a lot of problems to resolve on other projects. I will take a look as soon as I can.

**Comment by shymega** [comment]
@metal3d Thanks! Feel free to give me a prod once you have taken a look. Would be useful to figure out the root cause for v3.

**Comment by bennypowers** [comment]
I have this same issue with GNOME Wayland using patched input-leap and libportal (see #1728), should I open a new issue?

**Comment by shymega** [comment]
It's up to you. I personally would keep it to this issue, as it could be present on both backends.
